autogluon/utils/tabular/ml/augmentation/distill_utils.py

```diff
@@ -0,0 +1,177 @@
+import copy, time, traceback, logging, os, gc
+import numpy as np
+import pandas as pd
+from sklearn.neighbors import NearestNeighbors
+
+from ..constants import BINARY, MULTICLASS, REGRESSION, SOFTCLASS
+from ..models.tabular_nn.tabular_nn_model import TabularNeuralNetModel
+from ...metrics import mean_squared_error
+
+logger = logging.getLogger(__name__)
+
+def format_distillation_labels(y_train, y_test, problem_type, num_classes=None, eps_labelsmooth=0.01):
+    """ Transforms train/test label objects to the correct type for distillation.
+        eps_labelsmooth : truncates labels to [EPS, 1-EPS], eg. when converting binary problems -> regression
+    """
+    if problem_type == MULTICLASS:
+        y_train_int = y_train.to_numpy()
+        y_train = np.zeros((y_train_int.size, num_classes))
+        y_train[np.arange(y_train_int.size),y_train_int] = 1
+        y_train = pd.DataFrame(y_train)
+        y_test_int = y_test.to_numpy()
+        y_test = np.zeros((y_test_int.size, num_classes))
+        y_test[np.arange(y_test_int.size),y_test_int] = 1
+        y_test = pd.DataFrame(y_test)
+    elif problem_type == BINARY:
+        min_pred = 0.0
+        max_pred = 1.0
+        y_train = eps_labelsmooth + ((1-2*eps_labelsmooth)/(max_pred-min_pred)) * (y_train - min_pred)
+        y_test = eps_labelsmooth + ((1-2*eps_labelsmooth)/(max_pred-min_pred)) * (y_test - min_pred)
+
+    return (y_train, y_test)
+
+def augment_data(X_train, feature_types_metadata, augmentation_data=None, augment_method='spunge', augment_args={}):
+    """ augment_method options: ['spunge', 'munge']
+    """
+    if augmentation_data is not None:
+        X_aug = augmentation_data
+    else:
+        if 'num_augmented_samples' not in augment_args:
+            if 'max_size' not in augment_args:
+                augment_args['max_size'] = np.inf
+            augment_args['num_augmented_samples'] = int(min(augment_args['max_size'], augment_args['size_factor']*len(X_train)))
+
+        if augment_method == 'spunge':
+            X_aug = spunge_augment(X_train, feature_types_metadata, **augment_args)
+        elif augment_method == 'munge':
+            X_aug = munge_augment(X_train, feature_types_metadata, **augment_args)
+        else:
+            raise ValueError(f"unknown augment_method: {augment_method}")
+
+    return postprocess_augmented(X_aug, X_train)
+
+def postprocess_augmented(X_aug, X):
+    X_aug = pd.concat([X, X_aug])
+    # X_aug.drop_duplicates(keep='first', inplace=True)  # remove duplicate points including those in original training data already.
+    # TODO: dropping duplicates is much more efficient, but may skew distribution for entirely-categorical data with few categories.
+    X_aug = X_aug.tail(len(X_aug)-len(X))
```

Comment:
Ya that `postprocess_augmented` function was pretty much doing nothing with the duplicate-removal commented out. So I've commented out the call to the `postprocess_augmented` function, but left the original function in case we decide to go back to duplicate-removal in the future.

I'm 50-50 on duplicate removal, it seems like it would usually help, but may harm some all-categorical datasets by skewing the augmented data distribution.